### PR TITLE
Add "type" to membership queue metrics

### LINKED ISSUE #297: Membership Sync - Error: Added label "type" is not included in initial labelset: [ 'errcode', 'http_status' ]

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #297

### CODE REVIEW

File: src/components/membership-cache.ts [hunk 1]
```diff
@@ -103,7 +103,7 @@ export class MembershipCache {
             return Object.keys(this.membershipMap[roomId]);
         }
         const members = [];
-        for (const [userId, [membership, profile]] of Object.entries(this.membershipMap[roomId])) {
```
Comment (Half-Shot) [inline src/components/membership-cache.ts:106]: Random linter error I spotted

Comment (jaller94) [reply src/components/membership-cache.ts:106]: Why was this an error?
The value of `profile` isn't used.

File: src/components/membership-queue.ts [hunk 2]
```diff
@@ -131,7 +131,7 @@ export class MembershipQueue {
         this.failureReasonCounter = metrics.addCounter({
             name: "membershipqueue_reason",
             help: "Count of failures to process membership, by matrix errcode and http status",
```
Comment (jaller94) [inline src/components/membership-queue.ts:133]: This help lists all labels and should be updated (or have the list removed).

File: changelog.d/300.bugfix [hunk 3]
```diff
@@ -0,0 +1 @@
+Fix a bug that would cause membership queue failures to not be reported as metrics
```
Comment (jaller94) [inline changelog.d/300.bugfix:1]: I don't understand what this means. Why are failures reported "as metrics"? Do you mean "as part of the metrics"?
I also don't understand how adding the label "type" fixes some failures not being reported.

**Review by jaller94 — CHANGES_REQUESTED** [review]

File: changelog.d/300.bugfix (hunk 3 above)
Comment (jaller94) [reply changelog.d/300.bugfix:1]: Update about 2): After having read the issue I found the code where "type" is being used as a label later.
I now understand that this is about handing over all labels on the initial sync of metrics.

File: src/components/membership-cache.ts (hunk 1 above)
Comment (Half-Shot) [reply src/components/membership-cache.ts:106]: Well, warning. The linter wasn't happy that profile wasn't used.

File: changelog.d/300.bugfix (hunk 3 above)
Comment (Half-Shot) [reply changelog.d/300.bugfix]: Sorta. Prom-client will get angry if you use a label not defined in the schema. This PR just ensures we define it. Have updated the changelog entry to make it a bit easier to see the bug?

File: src/components/membership-queue.ts (hunk 2 above)
Comment (Half-Shot) [reply src/components/membership-queue.ts]: Added type.

File: src/components/membership-cache.ts (hunk 1 above)
Comment (jaller94) [reply src/components/membership-cache.ts:106]: I also read this the commit wrong way, thinking that `profile` got added.
Nevermind. :see_no_evil:

**Review by jaller94 — APPROVED** [review]
Thanks for clarifying the change in the news file.

**Pull request merged**